go/tosca/interpreter_registry.go

```diff
@@ -30,14 +30,47 @@ import (
 // GetInterpreter performs a lookup for the given name (case-insensitive) in
 // the registry. The result is nil if no interpreter was registered under the
 // given name.
+//
+// Deprecated: Use NewInterpreter instead.
 func GetInterpreter(name string) Interpreter {
+	res, err := NewInterpreter(name, nil)
+	if err != nil {
+		return nil
+	}
+	return res
+}
+
+// NewInterpreter performs a lookup for the given name (case-insensitive) in
+// the registry and creates a new Interpreter using the given optional
+// configuration. If no configuration is provided, the implementation uses
+// its default configuration. An error is returned if no factory was
+// registered under the given name.
+func NewInterpreter(name string, config ...any) (Interpreter, error) {
+	if len(config) > 1 {
+		return nil, fmt.Errorf("invalid configuration: too many arguments")
+	}
+	factory := GetInterpreterFactory(name)
+	if factory == nil {
+		return nil, fmt.Errorf("interpreter not found: %s", name)
+	}
+	c := any(nil)
+	if len(config) > 0 {
+		c = config[0]
```

Comment:
Since it is an any value. It would be possible to pass all the variadic arguments as a slice. 
Nevertheless, this feature is never used. Dead code?

Reply:
As said in the description, this feature introduces the option to pass configuration parameters like cache sizes to the interpreters. Not used know, but when we need it, we do not have to break the existing API.